app/jobs/foi_attachment_mask_job.rb

```diff
@@ -0,0 +1,43 @@
+##
+# Job to apply masks and censor rules to FoiAttachment objects. Masked file will
+# be stored as FoiAttachment#masked_file ActiveStorage association.
+#
+# Example:
+#   FoiAttachmentMaskJob.perform(FoiAttachment.first)
+#
+class FoiAttachmentMaskJob < ApplicationJob
+  queue_as :default
+  unique :until_and_while_executing, on_conflict: :log
+
+  attr_reader :attachment
+
+  delegate :incoming_message, to: :attachment
+
+  def perform(attachment)
+    @attachment = attachment
+
+    body = AlaveteliTextMasker.apply_masks(
```

Comment:
Since we've merged `#masked_file` into `#file` I'm going to keep this in a job, at least for now, as having this in `FoiAttachment#body` isn't really optimal as this is called throughout the codebase already - eg indexing, zip downloads. These places currently do there own masking but in the future we can refactor this.

Reply:
I didn't necessarily mean to inject it into the current method call chain; I just meant extract the logic into `FoiAttachment` so that it's not as buried inside the job:

```rb
class FoiAttachmentMaskJob
  def perform(attachment)
    attachment.apply_masks
  end
end

class FoiAttachment
  # `self` calls just included for clarity where they replace `attachment` from
  # the current implementation within `FoiAttachmentMaskJob`
  #
  # This only gets called via `FoiAttachmentMaskJob`
  def apply_masks
    body = AlaveteliTextMasker.apply_masks(
      self.unmasked_body,
      self.content_type,
      masks
    )

    if attachment.content_type == 'text/html'
      body =
        Loofah.scrub_document(body, :prune).
        to_html(encoding: 'UTF-8').
        try(:html_safe)
    end

    self.update(body: body, masked_at: Time.zone.now)
  end
end
```

Can do this later, but just wanted to clarify.